src/pages/editProfile/index.js

```diff
@@ -0,0 +1,244 @@
+import { useState, useEffect } from "react";
+import { useParams } from "react-router-dom";
+import TextInput from "../../components/form/textInput";
+import Button from "../../components/button";
+import ProfileCircle from "../../components/profileCircle";
+import Card from "../../components/card";
+import { useNavigate } from "react-router-dom";
+import useModal from "../../hooks/useModal";
+import ErrorMessage from "../../components/errorMessage";
+import "./style.css";
+import SaveChangesModal from "../../components/saveChangesModal"
+import { get } from "../../service/apiClient";
+import { emptyProfile } from "../../service/mockData";
+import useAuth from "../../hooks/useAuth";
+import ProfileImg from "../../components/profileImage/ProfileImage";
+
+
+const EditProfile = () => {
+  const [profile, setProfile] = useState(emptyProfile);
+  const [formState, setFormState] = useState(emptyProfile);
+  const [readOnly, setReadOnly] = useState(false)
+  const [passwordPermission, setPasswordPermission] = useState(false)
+  const [isError, setIsError] = useState(false);
+  const [noPassword, setNopassword] = useState(false)
```

Comment:
Currently, the default is `noPassword = false` which means there is a password, which doesn't feel like the right default.

Rather than dealing with double-negatives, it is usually easier to have state (and variables in genera) be phrased "positively", so this could instead be `hasUserEnteredPassword` with a default of `false`.